pkg/kubefed2/federate/util.go

```diff
@@ -106,24 +108,36 @@ func LookupAPIResource(config *rest.Config, key, targetVersion string) (*metav1.
 				lowerKey == strings.ToLower(resource.Kind) ||
 				lowerKey == fmt.Sprintf("%s.%s", resource.Name, gv.Group) {
 
-				targetResource = &resource
-				break
+				if targetResource != nil {
+					multiMatched = true
+				} else {
+					targetResource = resource.DeepCopy()
+					targetResource.Group = gv.Group
+					targetResource.Version = gv.Version
+				}
+
+				message += fmt.Sprintf(", %s.%s", resource.Name, gv.Group)
```

Comment:
Consider collecting matched resource names in a slice and using `strings.Join` to compose a message from the slice.